src/libs/Navigation/AppNavigator/createCustomBottomTabNavigator/TopBar.tsx

```diff
@@ -66,6 +75,10 @@ function TopBar({breadcrumbLabel, activeWorkspaceID, shouldDisplaySearch = true,
                         accessibilityLabel={translate('common.cancel')}
                         style={[styles.textBlue]}
                         onPress={() => {
+                            if (narrowSearchRouterActive) {
+                                deactivateNarrowSearchRouter?.();
+                                return;
+                            }
```

Comment:
If we have to now add 2 more attributes specifically for Search, then I would rather inject the whole callback `onSearchCancelPress` or simply `onCancelPress`

Cos right now a very generic `TopBar` component has 6 props, and 4 of them are search related :/

the naming is also all over the place:
```
    shouldDisplaySearch = true,
    shouldDisplayCancelSearch = false,
    narrowSearchRouterActive = false,
    deactivateNarrowSearchRouter,
```